### ISSUE TITLE
only collect slow requests

### ISSUE THREAD

**Issue by owexroasia** [appsignal/appsignal-ruby#605]
we are getting millions of requests for each user interaction. and all these data is not useful.
it's nice if there were configuration to filter out fast requests and only collect slow requests.
ex:
```yaml
log_slow_actions: 300ms
```
so it will only log any action that take more than given time period.

**Comment by roytomeij** [comment]
Hi @owexroasia! Unfortunately, that would impact the accuracy of our statistics, making a feature like Anomaly Detection less useful.

For instance, by filtering out fast requests, we wouldn't be able to tell you which interactions were slow, because we wouldn't be able to determine what your percentiles would be (put simply, if we don't know "normal", we can't know "not normal"). This would make it impossible to be notified of anomalies.

We do make sure not to overload you with samples though, and only show the slowest sample for minute per action.

**Comment by tombruijn** [comment]
We also replied in Intercom (our support system). Closing the issue here.
Let us know if you have any questions about this topic in the Intercom conversation. https://app.intercom.com/a/apps/yzor8gyw/inbox/inbox/540654/conversations/27121844510

**Comment by coorasse** [comment]
I also asked for this feature. Sure, stats would not work anymore, but for some of us those numbers are not that important.
I go to Appsignal to check my slow endpoints and I don't care about the overall statistics. Also, ignoring specific actions already makes those numbers biased anyway. 

Another point: when my applications grow I like sentry parameter `traces_sample_rate` that I can set to 0.1 and trace only 10% of all my requests. This keeps stats relevant and does not make my costs explode at the same time.

**Comment by roytomeij** [comment]
Hi @coorasse!

I’ve edited your comment to remove the code that samples requests. It’s a practice we don’t want to promote because sampling seriously diminishes AppSignal’s usability and reliability. We don’t want other customers to copy and paste this code and then reach out to our support engineers, asking why X doesn’t work as expected.

When sampling the way you do, you’re missing out on 90% of errors (assuming an even distribution of errors across requests). That doesn’t seem desirable. Apart from that, statistical information becomes incorrect because your 10% won’t be evenly distributed over fast/slow requests. This, in turn, makes Anomaly Detection fail because it relies on the same metrics. That then trickles down to notifiers such as PagerDuty, waking people up when it's not needed, and the other way around.

Long story short: AppSignal takes a holistic approach to error reporting and APM, providing context by tying everything together, which means that "nothing works" (properly) when introducing sampling at the top. That may work for you, but it's not how we envision AppSignal to work. If you mean to save money, feel free to reach out to support@appsignal.com to discuss how we can help :)

**Comment by paulRbr** [comment]
Hi @roytomeij, thanks for all the details. 

> Long story short: AppSignal takes a holistic approach to error reporting and APM,

Does this mean the gem traces 100% of the transactions?

I've tried to analyze the code to understand the decision you make to sample or not the data, but it seems that code is not open-source as I didn't find the definition of [the `appsignal_finish_transaction` C function](https://github.com/appsignal/appsignal-ruby/blob/8687732085202e5a1e4513f04c5b7163b32aaa23/ext/appsignal_extension.c#L296)
